Dockerfile

```diff
@@ -0,0 +1,244 @@
+# RUN curl --silent --show-error --location --fail --retry 3 --output /tmp/apache-ant.tar.gz https://archive.apache.org/dist/ant/binaries/apache-ant-1.10.5-bin.tar.gz \
+#    && tar xf /tmp/apache-ant.tar.gz -C /opt/ \
+#    && ln -s /opt/apache-ant-* /opt/apache-ant \
+#    && rm -rf /tmp/apache-ant.tar.gz \
+#    && /opt/apache-ant/bin/ant -version
+
+# ENV ANT_HOME=/opt/apache-ant
+
+# Install Gradle Version: 5.0
+# RUN curl --silent --show-error --location --fail --retry 3 --output /tmp/gradle.zip https://services.gradle.org/distributions/gradle-5.0-bin.zip \
+#    && unzip -d /opt /tmp/gradle.zip \
+#    && rm /tmp/gradle.zip \
+#    && ln -s /opt/gradle-* /opt/gradle \
+#    && /opt/gradle/bin/gradle -version
+
+# Install sbt from https://circle-downloads.s3.amazonaws.com/circleci-images/cache/linux-amd64/sbt-latest.tgz
+#RUN curl --silent --show-error --location --fail --retry 3 --output /tmp/sbt.tgz https://circle-downloads.s3.amazonaws.com/circleci-images/cache/linux-amd64/sbt-latest.tgz \
+#    && tar -xzf /tmp/sbt.tgz -C /opt/ \
+#    && rm /tmp/sbt.tgz \
+#    && /opt/sbt/bin/sbt sbtVersion
+
+# Install openjfx
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends openjfx
+
+# Add build-essential
+RUN apt-get install -y build-essential
+
+# Add RISCV toolchain necessary dependencies
+RUN apt-get update
+RUN apt-get install -y \
+            autoconf \
+            automake \
+            autotools-dev \
+            babeltrace \
+            bc \
+            curl \
+            device-tree-compiler \
+            expat \
+            flex \
+            gawk \
+            gperf \
+            g++ \
+            libexpat-dev \
+            libgmp-dev \
+            libmpc-dev \
+            libmpfr-dev \
+            libtool \
+            libusb-1.0-0-dev \
+            make \
+            patchutils \
+            pkg-config \
+            python3 \
+            texinfo \
+            zlib1g-dev \
+            rsync \ 
+            bison \ 
+            verilator
+
+
+# Use specific bison version to bypass Verilator 4.034 issues
+# TODO: When Verilator is bumped, use apt to get newest bison
+# RUN wget https://ftp.gnu.org/gnu/bison/bison-3.5.4.tar.gz \
+#    && tar -xvf bison-3.5.4.tar.gz \
+#    && cd bison-3.5.4 \
+#    && ./configure && make && make install
+
+# Check bison version is 3.5.4
+# RUN bison --version
+
+# Add minimal QEMU dependencies
+RUN apt-get install -y \
+            libfdt-dev \
+            libglib2.0-dev \
+            libpixman-1-dev
+
+# Install verilator
+# RUN git clone http://git.veripool.org/git/verilator \
+#    && cd verilator \
+#    && git pull \
+#    && git checkout v4.034 \
+#    && autoconf && ./configure && make && make install
+
+
+# Add HOME environment variable
+ENV HOME="/home/riscvuser"
+
+# Update PATH for RISCV toolchain (note: hardcoded for CircleCI)
+ENV RISCV="$HOME/riscv-tools-install"
+ENV LD_LIBRARY_PATH="$RISCV/lib"
+ENV PATH="$RISCV/bin:$PATH"
```

Comment:
Do these still need to be set here?

Reply:
Since we are not currently sourcing env.sh, yes.  They ensure that when the Docker image is used, environment variables are set even without sourcing env.sh.  

I should probably fire up the image to make sure that CI defaults to using user riscvuser, and it may be preferable to just source env.sh when using this in CI.  

For users who aren't using CI, these env vars ensure that they don't need to source env.sh when using the image.  

**goal:** This Docker image should allow anyone to play with Chipyard quickly and easily, using the exact same image that we use in the Tendermint Hardware [CI system](https://gitlab.com/virgohardware/core/-/pipelines) so that:

* questions about differing build environments are eliminated
* time spent configuring a build environment is drastically reduced
* need to own/rent a build machine is eliminated

all of those were a big problems for me when I first began researching Open Source Semiconductors.  

* users should be able to modify a core, create a branch in our repo, and build/test it so that the pace of development, especially for non-professionals, increases

Reply:
I agree with the goals you specified.

I believe that setting `RISCV` here in the Dockerfile does not actually adjust the location of the eventual riscv-tools install, as the `build-toolchains.sh` hard-code a path (`{riscv/esp}-tools-install`) unless the `--prefix` argument is given. You will probably observe this when you inspect the final image.

Since Chipyard provides two toolchain options for users, I think its best to leave RISCV unset in final Dockerimage. The Chipyard makefiles will already provide a informative error if running RISCV unset, and the documentation additionally stresses that RISCV must be set by sourcing one of the `env` scripts.

Reply:
Thanks for your advice!

I'll take that out, and see what I find :).